bin/arrayNormalization.R

```diff
@@ -0,0 +1,374 @@
+			# 		- For ST arrays, a probe set =~ an exon.
+			#
+			# For ST arrays (ExonFeatureSet or GeneFeatureSet), the
+			# summarization target argument is needed:
+			# 	- target="core" obtains gene-level summaries of expression for
+			# 	"well-annotated" loci (annotated by RefSeq). 
+			# 	- Exon (not Gene) arrays also have probes for "predicted genes"
+			# 	and other less well-described loci. These can be accessed with
+			# 	target="extended" (mRNA evidence) or target="full"
+			# 	(bioinformatic prediction only).
+			# 	- To summarize at the exon level, e.g. to look at expressions
+			# 	for alternative splice forms, use target="probeset".
+			# 	- For now we want to display information for genes, we are not
+			# 	going as far as individual exons, so we'll use target="core".
+			#
+			# For non-ST arrays (ExpressionFeatureSet or SnpCnvFeatureSet), do
+			# not specify target (dies if you do).
+			#
+			# rma() creates an ExpressionSet object which we'll call eSet.
+			
+			# If the first element of the vector returned by class() matches
+			# "Gene" or "Exon", we've got an ST array...
+			if( grepl("Gene", class(featureSet)[1]) || grepl("Exon", class(featureSet[1]) )) {
+				eSet <- rma(featureSet, target="core")
+			} 
+			# ...otherwise, we haven't.
+			else {
+				eSet <- rma(featureSet) # no target
+			}
+			print("rma finished")
+		 
+		# Now the pre-processing is done, either with oligo or affy, and the
+		# data is stored in the eSet object. 
+	
+		# Next, make sure the scan names in the vector 'assayNames' are in the same
+		# order as their corresponding samples in eSet.
+		# shortFileNames is vector of CEL filenames with /path/to/files/ stripped off (i.e. just base names).
+		shortFileNames <- gsub(".+/", "", files)
+		
+		# Create a vector of scan names in the correct order with relSort().
+		assayNamesSorted <- relSort(assayNames, shortFileNames, sampleNames(eSet))
+		
+		# Normalized expressions to data frame.
+		normExprs <- exprs(eSet)
+		# Add row names as a column and give it col name "DesignElementAccession".
+		normExprs <- data.frame(cbind(rownames(normExprs), normExprs))
+	    colnames(normExprs) <- c("DesignElementAccession", assayNamesSorted)
+
+		print("Writing normalized expressions matrix")
+		# Write normalized data to file.
+		write.table(normExprs, file = outFile, sep = "\t", quote = FALSE, row.names=FALSE)
+		}
+	})
+
+	# error catching -- return the error message if there was one (class(error)
+	# returns "try-error"), otherwise return NULL.
+	if ("try-error" == class(error)) {
+		return(error)
+	} else {
+		return(NULL)
+	}
+}
+
+
+
+
+# subsetProbes()
+#  - For microRNA, we only want to normalize using probesets that are mapped to
+#  the latest release of miRBase (www.mirbase.org). Subset the data here.
+# ARGUMENTS:
+# 	- dataSet <- for now either EListRaw or RGList object.
+# 	- miRBaseFile <- filename for miRBase mappings.
+subsetProbes <<- function(dataSet, miRBaseFile) {
```

Comment:
extra < on this line (303)

Reply:
Good spot. Although this function works :)